packages/amplify-graphql-searchable-transformer/src/nodeToNodeEncryption.ts

```diff
@@ -15,32 +14,32 @@ import { printer } from '@aws-amplify/amplify-prompts';
  * @param apiName the name of the api to attempt and pull the flag from.
  * @returns whether or not NodeToNodeEncryption should be enabled on a searchable instance.
  */
-export const shouldEnableNodeToNodeEncryption = (apiName: string): boolean => {
+export const shouldEnableNodeToNodeEncryption = (apiName: string): { enabled: boolean, log: string | undefined } => {
   try {
     const nodeToNodeEncryptionParameter = getNodeToNodeEncryptionConfigValue(apiName);
     const doesExistingBackendHaveNodeToNodeEncryption = getCurrentCloudBackendStackFiles(apiName).some(definition => hasNodeToNodeEncryptionOptions(definition));
 
-    warnOnExistingNodeToNodeEncryption(doesExistingBackendHaveNodeToNodeEncryption);
+    const log = warnOnExistingNodeToNodeEncryption(doesExistingBackendHaveNodeToNodeEncryption);
 
     if (nodeToNodeEncryptionParameter !== undefined) {
-      return nodeToNodeEncryptionParameter;
+      return { enabled: nodeToNodeEncryptionParameter, log };
     }
 
-    return doesExistingBackendHaveNodeToNodeEncryption;
+    return { enabled: doesExistingBackendHaveNodeToNodeEncryption, log };
   } catch (e) {
     // Fail open, and don't set the flag for the purposes of this workaround phase.
-    return false;
+    return { enabled: false, log: undefined };
```

Comment:
Should the log be set with the error msg in this case?

Reply:
I was trying to keep functionality the same with the refactor. I can update this with a later PR though.